src/main/java/com/bio4j/angulillos/TypedElement.java

```diff
@@ -44,7 +44,13 @@
 
     Element types are also used as factories for constructing instances of the corresponding elements.
   */
-  interface Type {
+  interface Type <
+    E extends TypedElement<E,ET,G,ER>,
+    ET extends TypedElement.Type<E,ET,G,ER>,
+    G extends TypedGraph<G,?,?>,
+    ER
+  >
+  {
```

Comment:
Can you tell me, where do you use `TypedElement.Type` with all those parameters? I don't see any place. It's just a base trait to add this `name()` to both vertex type and edge type.